Mastodon/Scene/Onboarding/Login/MastodonLoginViewController.swift

```diff
@@ -139,30 +139,23 @@ class MastodonLoginViewController: UIViewController, NeedsDependency {
     
     @objc func login() {
         guard let server = viewModel.selectedServer else { return }
-        
+
         authenticationViewModel
-            .authenticated
-            .asyncMap { domain, user -> Result<Bool, Error> in
-                do {
-                    let result = try await self.context.authenticationService.activeMastodonUser(domain: domain, userID: user.id)
-                    return .success(result)
-                } catch {
-                    return .failure(error)
-                }
-            }
-            .receive(on: DispatchQueue.main)
-            .sink { [weak self] result in
-                guard let self = self else { return }
-                switch result {
-                case .failure(let error):
-                    assertionFailure(error.localizedDescription)
-                case .success(let isActived):
-                    assert(isActived)
-                    self.coordinator.setup()
+            .authenticated.sink { (domain, account) in
+                Task {
+                    do {
+                        _ = try await self.context.authenticationService.activeMastodonUser(domain: domain, userID: account.id)
+                        FileManager.default.store(account: account, forUserID: MastodonUserIdentifier(domain: domain, userID: account.id))
+                        Task { @MainActor in
+                            self.coordinator.setup()
```

Comment:
This feels a bit weird, nesting a `Task` inside a Combine sink, then spawning yet another `Task`.
I actually really like the `asyncMap` approach, and maybe we could annotate `coordinator.setup()` with `@MainActor` so we can get rid of the second `Task` her?